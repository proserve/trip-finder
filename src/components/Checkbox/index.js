import React from 'react';
import PropTypes from 'prop-types';

import './checkbox.scss';

export default class Checkbox extends React.Component {
  render() {
    const { isChecked, onChange, name } = this.props;
    return (
      <label htmlFor={`${name}_checkbox`} className="container-checkbox">
        {name}
        <input
          type="checkbox"
          id={`${name}_checkbox`}
          checked={isChecked}
          onChange={onChange}
        />
        <span className="checkmark" />
      </label>
    );
  }
}

Checkbox.propTypes = {
  isChecked: PropTypes.bool,
  onChange: PropTypes.func.isRequired,
  name: PropTypes.string.isRequired,
};

Checkbox.defaultProps = {
  isChecked: false,
};
