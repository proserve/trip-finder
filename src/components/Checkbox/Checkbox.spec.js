import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from './index';

describe('Checkbox', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Checkbox name="test" onChange={() => { console.log('test'); }} />);
    expect(component).toMatchSnapshot();
  });
});
