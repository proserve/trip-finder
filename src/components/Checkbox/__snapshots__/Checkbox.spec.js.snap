// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render correctly in "debug" mode 1`] = `
<label
  className="container-checkbox"
  htmlFor="test_checkbox"
>
  test
  <input
    checked={false}
    id="test_checkbox"
    onChange={[Function]}
    type="checkbox"
  />
  <span
    className="checkmark"
  />
</label>
`;
